flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -428,11 +434,29 @@ DispatcherResourceManagerComponentFactory createDispatcherResourceManagerCompone
 	}
 
 	@VisibleForTesting
-	protected HighAvailabilityServices createHighAvailabilityServices(Configuration configuration, Executor executor) throws Exception {
+	protected HighAvailabilityServices createHighAvailabilityServices(
+			Configuration configuration,
+			Executor executor) throws Exception {
 		LOG.info("Starting high-availability services");
-		return HighAvailabilityServicesUtils.createAvailableOrEmbeddedServices(
-			configuration,
-			executor);
+		return miniClusterConfiguration.embeddedHaLeadershipControlEnabled() ?
+			new EmbeddedHaServicesWithLeadershipControl(executor) :
+			HighAvailabilityServicesUtils.createAvailableOrEmbeddedServices(configuration, executor);
+	}
+
+	/**
+	 * Returns {@link HaLeadershipControl} if enabled.
+	 *
+	 * <p>{@link HaLeadershipControl} allows granting and revoking leadership of HA components,
+	 * e.g. JobManager. The method return {@link Optional#empty()} if the control is not enabled in
+	 * {@link MiniClusterConfiguration}.
+	 *
+	 * <p>Enabling this feature disables {@link HighAvailabilityOptions#HA_MODE} option.
```

Comment:
This paragraph should be added to `enableHaLeadershipControl` method on the builder, I guess.